### PR TITLE
Implement general state cheatcodes

### PR DESCRIPTION
**Pull request by anvacaru**
Part of: https://github.com/runtimeverification/foundry/issues/12

### CODE REVIEW

**Comment by ehildenb** [comment]
It looks good, we need to wait for https://github.com/runtimeverification/evm-semantics/pull/1332 to fix the `failed` vs `_failed` issue.

File: foundry.md [hunk 1]
```diff
@@ -129,6 +129,106 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
...
 ```
 
+#### `warp` - Sets the block timestamp.
+
+```
+function warp(uint256) external;
+```
+
+`call.warp` will match when the `warp` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the uint256 value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<timestamp>` cell.
```
Comment (tothtamas28) [inline foundry.md:139]:
```suggestion
This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<timestamp>` cell.
```

File: foundry.md [hunk 2]
```diff
@@ -129,6 +129,106 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
...
+```
+
+#### `roll` - Sets the block number.
+
+```
+function roll(uint256) external;
+```
+
+`call.roll` will match when the `roll` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the uint256 value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<number>` cell.
```
Comment (tothtamas28) [inline foundry.md:159]:
```suggestion
This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<number>` cell.
```

File: foundry.md [hunk 3]
```diff
@@ -129,6 +129,106 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
...
+```
+
+#### `fee` - Sets the block number.
+
+```
+function fee(uint256) external;
+```
+
+`call.fee` will match when the `fee` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the uint256 value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<baseFee>` cell.
```
Comment (tothtamas28) [inline foundry.md:179]:
```suggestion
This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<baseFee>` cell.
```

File: foundry.md [hunk 4]
```diff
@@ -129,6 +129,106 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
...
+```
+
+#### `chainId` - Sets the block number.
+
+```
+function chainId(uint256) external;
+```
+
+`call.chainId` will match when the `chanId` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the uint256 value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<chainID>` cell.
```
Comment (tothtamas28) [inline foundry.md:199]:
```suggestion
This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<chainID>` cell.
```

File: foundry.md [hunk 5]
```diff
@@ -129,6 +129,106 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
...
+```
+
+#### `coinbase` - Sets the block number.
+
+```
+function coinbase(address) external;
+```
+
+`call.coinbase` will match when the `coinbase` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the uint256 value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<coinbase>` cell.
```
Comment (tothtamas28) [inline foundry.md:219]:
```suggestion
This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<coinbase>` cell.
```

**Review by tothtamas28 — APPROVED** [review]
LGTM, great work!

File: foundry.md [hunk 6]
```diff
@@ -116,7 +116,7 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
          <output> _ => .ByteArray </output>
          <localMem> LM </localMem>
       requires CHEAT_ADDR ==Int #address(FoundryCheat)
-       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 3364511341 // selector ( "deal" )
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 3364511341 // selector ( "deal(address,uint256)" )
       [priority(40)]
```
Comment (ehildenb) [inline foundry.md:121]: Can we also reformat `call.deal` rule to have `<k>` cell on one line too?

Comment (anvacaru) [reply foundry.md:117]: Imo, as a one-liner it has decreased readability

**Pull request merged**